module/zfs/spa.c

```diff
@@ -455,11 +456,8 @@ spa_prop_get(spa_t *spa, nvlist_t **nvp)
 				src = ZPROP_SRC_LOCAL;
 
 			if (prop == ZPOOL_PROP_BOOTFS) {
-				dsl_pool_t *dp;
 				dsl_dataset_t *ds = NULL;
 
-				dp = spa_get_dsl(spa);
-				dsl_pool_config_enter(dp, FTAG);
 				err = dsl_dataset_hold_obj(dp,
 				    za.za_first_integer, FTAG, &ds);
 				if (err != 0) {
```

Comment:
Nice find.  You're going to want to remove this extra `dsl_pool_config_exit` for this error case.

Reply:
Ah, thanks, fixed.